src/monattack.cpp

```diff
@@ -2717,7 +2722,122 @@ bool mattack::fear_paralyze( monster *z )
 
     return true;
 }
+bool mattack::nurse_assist( monster *z )
+{
+    if( !within_visual_range( z, 6 ) ) {
+        g->u.remove_effect( effect_assisted );
+        return false;
+    }
+    bool u_see = g->u.sees( *z );
+
+    if( u_see && one_in( 10 ) ) {
+        add_msg( m_info, _( "The %s is scanning it's surroundings." ), z->name() );
+    }
+
+    if( g->u.is_wearing( "badge_doctor" ) ||
+        z->attitude_to( g->u ) == monster_attitude::MATT_FRIEND ) {
+        sounds::sound( z->pos(), 8, sounds::sound_t::speech,
+                       string_format(
+                           _( "a soft robotic voice says, \"Welcome doctor %s.  I'll be your assistant today.\"" ),
+                           Name::generate( g->u.male ) ) );
+        g->u.add_effect( effect_assisted, 20_turns );
+    } else {
+        g->u.remove_effect( effect_assisted );
+    }
+}
+bool mattack::nurse_operate( monster *z )
+{
+    const std::string ammo_type( "anesthetic" );
+
+    if( z->has_effect( effect_dragging ) || z->has_effect( effect_operating ) ) {
+        return false;
+    }
+    bool u_see = g->u.sees( *z );
+
+    if( u_see && one_in( 10 ) ) {
+        add_msg( m_info, _( "The %s is scanning it's surroundings." ), z->name() );
+    }
+
+
+    if( ( ( g->u.is_wearing( "badge_doctor" ) ||
+            z->attitude_to( g->u ) == monster_attitude::MATT_FOLLOW ) && u_see ) && one_in( 30 ) ) {
 
+        add_msg( m_info, _( "The %s doesn't seem to register you as a doctor." ), z->name() );
+    }
+
+    if( z->ammo[ammo_type] == 0 && u_see && one_in( 30 ) ) {
+        add_msg( m_info, _( "The %s looks at its empty anesthesia kit with a dejected look." ), z->name() );
+        return false;
```

Comment:
Is it intended that nurse bot will cease to try to forcibly operate you only in 3% of cases, even with empty anesthesia kit?

Reply:
Nop,  the one_in is just for the message to appear less often. I missed the fact that it was also affecting the return. I'll fix it.